### ISSUE TITLE
Terraform provider does not work well when deploying a Flink Model/Statement that uses  sql.secrets.*

### ISSUE THREAD

**Issue by novarz** [confluentinc/terraform-provider-confluent#397]
TF tries to redeploy the resource even when nothing has changed with the consequent failure.

![image](https://github.com/confluentinc/terraform-provider-confluent/assets/7761976/ae00a4b5-ab7e-4ccb-baf0-d6aea344fe03)

**Comment by novarz** [comment]
Here is the TF 

# --------------------------------------------------------
# Flink SQL: CREATE Model vector_encoding
# --------------------------------------------------------
resource “confluent_flink_statement” “create_model” {
  depends_on = [
        resource.confluent_environment.environment,
        resource.confluent_schema_registry_cluster.essentials,
        resource.confluent_kafka_cluster.cluster,
        resource.confluent_connector.datagen_products,
        resource.[confluent_flink_compute_pool.my](http://confluent_flink_compute_pool.my/)_compute_pool,
        resource.confluent_role_binding.app-general-environment-admin
  ]
  organization {
    id = data.confluent_organization.main.id
  }
   environment {
    id = confluent_environment.environment.id
  }
  compute_pool {
    id = [confluent_flink_compute_pool.my](http://confluent_flink_compute_pool.my/)_compute_pool.id
  }
  principal {
    id = confluent_service_account.app-general.id
  }
   properties = {
    “sql.current-catalog”   : confluent_environment.environment.display_name
    “sql.current-database”  : confluent_kafka_cluster.cluster.display_name
    “sql.secrets.openaikey” : var.openai_key
  }
  statement  = “CREATE MODEL `vector_encoding` INPUT (input STRING) OUTPUT (vector ARRAY<FLOAT>) WITH( ‘TASK’ = ‘classification’,‘PROVIDER’ = ‘OPENAI’,‘OPENAI.ENDPOINT’ = ‘https://api.openai.com/v1/embeddings','OPENAI.API_KEY’ = ‘{{sessionconfig/sql.secrets.openaikey}}’);”
  rest_endpoint   =  data.[confluent_flink_region.my](http://confluent_flink_region.my/)_flink_region.rest_endpoint
  credentials {
    key    = [confluent_api_key.my](http://confluent_api_key.my/)_flink_api_key.id
    secret = [confluent_api_key.my](http://confluent_api_key.my/)_flink_api_key.secret
  }
  lifecycle {
    prevent_destroy = false
  }
}

**Comment by linouk23** [comment]
Thank you for creating this issue! It does seem like a bug to me, we'll take a look at it soon 👍

**Comment by ConfluentSpencer** [comment]
@tmalikconfluent can you take a look at this one?

**Comment by BlakeItOut** [comment]
I don't think this is the way API keys are handled anymore with Flink Connections now available so is this still applicable?

**Comment by linouk23** [comment]
> I don't think this is the way API keys are handled anymore with Flink Connections now available so is this still applicable?

Could you add more details to your question @BlakeItOut? Thank you!

update: we did merge a fix for this issue to our internal repository and we'll be copying it to this repository soon.

**Comment by BlakeItOut** [comment]
Based on the model inference docs it seems to point toward creating a Flink Connection for your model access creds instead of including it directly in your WITH options. It is unclear from the documentation if this other way of including  creds is even still allowed. "All of the CREATE MODEL statements require a connection resource that you create by using the confluent flink connection create command" from https://docs.confluent.io/cloud/current/flink/reference/statements/create-model.html#examples

**Comment by linouk23** [comment]
Flink Connection is going to be a separate resource in TF.

**Comment by linouk23** [comment]
@novarz we've released a fix in the most recent version of TF Provider (2.15.0), thank you for waiting!
